mmlearn/tasks/classification.py

```diff
@@ -0,0 +1,250 @@
+                def __init__(
+                    self, descriptions: List[str], tokenizer: PreTrainedTokenizerBase
+                ) -> None:
+                    self.descriptions = descriptions
+                    self.tokenizer = tokenizer
+
+                def __len__(self) -> int:
+                    return len(self.descriptions)
+
+                def __getitem__(self, idx: int) -> Example:
+                    description = self.descriptions[idx]
+                    tokens = self.tokenizer(description)
+
+                    example = Example(
+                        {
+                            Modalities.RGB: torch.rand(3, 224, 224),
+                            Modalities.TEXT: description,
+                        }
+                    )
+
+                    if tokens is not None:
+                        if isinstance(tokens, dict):  # output of HFTokenizer
+                            assert (
+                                Modalities.TEXT in tokens
+                            ), f"Missing key `{Modalities.TEXT}` in tokens."
+                            example.update(tokens)
+                        else:
+                            example[Modalities.TEXT] = tokens
+
+                    return example
+
+            if self.tokenizer is None:
+                raise ValueError(
+                    "Tokenizer must be set in the dataset to generate tokenized label descriptions"
+                )
+
+            # Set the label embeddings
+            for name, dataset_info in all_dataset_info.items():
+                descriptions = [
+                    "This image has a sign of " + label
+                    for label in dataset_info.get_label_mapping().values()
+                ]
+
+                dataset = LabelDescriptionDataset(descriptions, self.tokenizer)
+                batch_size = len(dataset)
+                dataloader = DataLoader(
+                    dataset,
+                    batch_size=batch_size,
+                    shuffle=False,
+                    collate_fn=collate_example_list,
+                )
+                batch = next(iter(dataloader))
+                batch = {
+                    key: value.to(pl_module.device) if torch.is_tensor(value) else value
+                    for key, value in batch.items()
+                }
+                all_dataset_info[name].set_label_embedding(
+                    pl_module(batch)[Modalities.get_modality(Modalities.TEXT).embedding]
+                )
+
+        self.all_dataset_info = all_dataset_info
+
+        for dataset_name in list(all_dataset_info.keys()):
+            for spec in self.task_specs:
+                assert Modalities.has_modality(spec.query_modality)
+                query_modality = Modalities.get_modality(spec.query_modality)
+                mode = self.mode
+                metric_name = spec.metric_name
+                metric = getattr(torchmetrics, metric_name)
```

Comment:
Instead of this, we can create configs for torchmetric objects (using hydra_zen) and instantiate them from the config.